src/components/input-time-picker/input-time-picker.e2e.ts

```diff
@@ -291,25 +375,160 @@ describe("calcite-input-time-picker", () => {
+    await page.keyboard.type("14:2:3");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(0);
+
+    await page.keyboard.press("Enter");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(1);
+    expect(await inputTimePicker.getProperty("value")).toBe("14:02:03");
+    expect(await getInputValue(page)).toBe("14:02:03");
+
+    await page.keyboard.press("Enter");
     await page.waitForChanges();
 
-    expect(changeEvent).toHaveReceivedEventTimes(2);
+    expect(changeEvent).toHaveReceivedEventTimes(1);
   });
 
-  it("formats valid typed time value appropriately on blur", async () => {
+  it("committing hh:mm value when step=1 with the Enter key works as expected for a 12-hour locale", async () => {
     const page = await newE2EPage();
-    await page.setContent(`<calcite-input-time-picker step="1"></calcite-input-time-picker><input>`);
+    await page.setContent(`<calcite-input-time-picker step="1"></calcite-input-time-picker>`);
+
+    const inputTimePicker = await page.find("calcite-input-time-picker");
+    const changeEvent = await inputTimePicker.spyOnEvent("calciteInputTimePickerChange");
+
+    expect(changeEvent).toHaveReceivedEventTimes(0);
+
+    await page.keyboard.press("Tab");
+    await page.keyboard.type("5:4 PM");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(0);
+
+    await page.keyboard.press("Enter");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(1);
+    expect(await inputTimePicker.getProperty("value")).toBe("17:04:00");
+    expect(await getInputValue(page)).toBe("05:04:00 PM");
+
+    await page.keyboard.press("Enter");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(1);
+  });
+
+  it("committing hh:mm value when step=1 with the Enter key works as expected for a 24-hour locale", async () => {
+    const page = await newE2EPage();
+    await page.setContent(`<calcite-input-time-picker step="1" lang="fr"></calcite-input-time-picker>`);
+
+    const inputTimePicker = await page.find("calcite-input-time-picker");
+    const changeEvent = await inputTimePicker.spyOnEvent("calciteInputTimePickerChange");
+
+    expect(changeEvent).toHaveReceivedEventTimes(0);
+
+    await page.keyboard.press("Tab");
+    await page.keyboard.type("14:2");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(0);
+
+    await page.keyboard.press("Enter");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(1);
+    expect(await inputTimePicker.getProperty("value")).toBe("14:02:00");
+    expect(await getInputValue(page)).toBe("14:02:00");
+
+    await page.keyboard.press("Enter");
+    await page.waitForChanges();
+
+    expect(changeEvent).toHaveReceivedEventTimes(1);
+  });
+
+  it("attempting to commit an invalid time value works as expected", async () => {
```

Comment:
Could you add more info to the test name to describe that the invalid value won't be committed, but the input is left as is?

Reply:
good suggestion.  Updated it to read "attempting to commit an invalid time value fails, but leaves the typed value intact"